otter/test/test_cloud_client.py

```diff
@@ -255,6 +269,199 @@ def test_params(self):
+        deferred_lock.side_effect = DeferredLock
+
+        clock = Clock()
+        bracket = _serialize_and_delay(clock, 15)
+
+        result = bracket(lambda: succeed('foo'))
+        clock.advance(14)
+        self.assertNoResult(result)
+        clock.advance(15)
+        self.assertEqual(self.successResultOf(result), ('locked', 'foo'))
+
+
+class DefaultThrottlerTests(SynchronousTestCase):
+    """Tests for :func:`_default_throttler`."""
+
+    def test_mismatch(self):
+        """policy doesn't have a throttler for random junk."""
+        bracket = _default_throttler(None, 'foo', 'get')
+        self.assertIs(bracket, None)
+
+    def test_post_cloud_servers(self):
+        """POSTs to cloud servers get throttled by a second."""
+        clock = Clock()
+        bracket = _default_throttler(clock, ServiceType.CLOUD_SERVERS, 'post')
+        d = bracket(lambda: 'foo')
+        self.assertNoResult(d)
+        clock.advance(1)
+        self.assertEqual(self.successResultOf(d), 'foo')
+
+    def test_delete_cloud_servers(self):
+        """DELETEs to cloud servers get throttled by a second."""
+        clock = Clock()
+        bracket = _default_throttler(clock,
+                                     ServiceType.CLOUD_SERVERS, 'delete')
+        d = bracket(lambda: 'foo')
+        self.assertNoResult(d)
+        clock.advance(1)
+        self.assertEqual(self.successResultOf(d), 'foo')
+
+    def test_post_and_delete_not_the_same(self):
+        """
+        The throttlers for POST and DELETE to cloud servers are different.
+        """
+        clock = Clock()
+        deleter = _default_throttler(clock, ServiceType.CLOUD_SERVERS,
+                                     'delete')
+        poster = _default_throttler(clock, ServiceType.CLOUD_SERVERS, 'post')
+        self.assertIsNot(deleter, poster)
+
+    def test_post_delay_configurable(self):
+        """The delay for creating servers is configurable."""
+        set_config_data(
+            {'cloud_client': {'throttling': {'create_server_delay': 500}}})
+        self.addCleanup(set_config_data, {})
+        clock = Clock()
+        bracket = _default_throttler(clock, ServiceType.CLOUD_SERVERS, 'post')
+        d = bracket(lambda: 'foo')
+        clock.advance(499)
+        self.assertNoResult(d)
+        clock.advance(500)
+        self.assertEqual(self.successResultOf(d), 'foo')
+
+    def test_delete_delay_configurable(self):
+        """The delay for deleting servers is configurable."""
+        set_config_data(
+            {'cloud_client': {'throttling': {'delete_server_delay': 500}}})
+        self.addCleanup(set_config_data, {})
+        clock = Clock()
+        bracket = _default_throttler(clock,
+                                     ServiceType.CLOUD_SERVERS, 'delete')
+        d = bracket(lambda: 'foo')
+        clock.advance(499)
+        self.assertNoResult(d)
+        clock.advance(500)
+        self.assertEqual(self.successResultOf(d), 'foo')
+
```

Comment:
No. Its fine to have test there. I was just wondering if you can have a serialization check that doesn't mock `DeferredLock` but I guess it really doesn't make much difference. Its fine :)